components/server/ee/src/user/stripe-service.ts

```diff
@@ -98,7 +99,22 @@ export class StripeService {
         return priceInformation.humanReadableDescription;
     }
 
-    async cancelSubscription(subscriptionId: string): Promise<void> {
+    async cancelSubscriptionForUser(userId: string) {
+        let subscriptionId;
+        try {
+            subscriptionId = await this.findUncancelledSubscriptionByAttributionId(
+                AttributionId.render({ kind: "user", userId }),
+            );
+            if (subscriptionId) {
+                await this.cancelSubscription(subscriptionId);
+            }
+        } catch (error) {
+            log.error("Error cancelling Stripe user subscription", error, { subscriptionId });
+            throw new Error(`Failed to cancel stripe subscription. ${error}`);
+        }
+    }
+
+    protected async cancelSubscription(subscriptionId: string): Promise<void> {
```

Comment:
Ah, this is a remnant of our split brain dealing with Stripe. 

On usage, there's `CancelSubscription` which would ideally handle every detail of this, such that it can be uniformly called.

Reply:
Yep, totally agree that we should fold this code into the usage side.
So far I've refrained because I'm waiting for authN. But we could also do it earlier, too! Maybe out of scope for this PR, though.